src/core/qgsmaplayer.cpp

```diff
@@ -306,20 +302,6 @@ bool QgsMapLayer::readLayerXml( const QDomElement &layerElement,  QgsReadWriteCo
     }
   }
 
-  // use scale dependent visibility flag
```

Comment:
These changes - while desirable - make me nervous. There's potentially signals getting thrown in the moved code and the change of property restoration order here may have unintentional tricky side effects. I don't think that should block this refactoring... But proceed with extreme caution!